src/comments.js

```diff
@@ -834,7 +834,33 @@ function printComment(commentPath, options) {
       return "#" + comment.value;
     case "CommentBlock":
     case "Block":
-      return "/*" + comment.value + "*/";
+      const lines = comment.value.split("\n");
+      const isJsDocComment =
+        lines.length > 1 &&
+        lines.slice(0, lines.length - 1).every(line => line.trim()[0] === "*");
+      if (!isJsDocComment) {
+        return "/*" + comment.value + "*/";
+      }
+
+      // Determine the indentation level of the parent node.
+      const parentNode = commentPath.getParentNode();
+      const indentLevel = parentNode.loc.start.column;
```

Comment:
You got me thinking about this and I don't think that checking the parent is the right approach after-all. If we look at what happens today, the first line of the comment has the right indentation, what's problematic are the remaining lines.

Instead of sending a raw string for the entire comment, you could send one string for each line with `hardline` in between. The question now is: how many spaces do I need to prepend each line such that when it's aligned the same as in the original line, it's going to be printed as the right position.

Solving this is easy, it's the position of the first non-whitespace - position of the beginning of the first line. You can use `align()` to specify this number for each line.